app/models/vote.rb

```diff
@@ -0,0 +1,24 @@
+
+
+class Vote < ApplicationRecord
+
+  belongs_to :user
+  belongs_to :work
+
+  validates :user_id, presence: true
+  validates :work_id, presence: true
+
+  validate :one_vote_per_user_per_work
+
+  def one_vote_per_user_per_work
```

Comment:
Neat custom validation.  You can also do something similar with `uniqueness` and `scope` like in the [rails api guide](http://guides.rubyonrails.org/active_record_validations.html#uniqueness)